include/uart.h

```diff
@@ -28,6 +29,118 @@ extern "C" {
 
 #include <device.h>
 
+#ifdef CONFIG_UART_HIGH_LEVEL_API
+
+/** @brief Possible configuration settings */
+enum uart_config {
+	UART_CFG_BAUDRATE,
+	UART_CFG_PARITY,
+	UART_CFG_STOP_BITS,
+	UART_CFG_DATA_BITS,
+	UART_CFG_FLOW_CONTROL,
+	UART_CFG_CALLBACK,
+	UART_CFG_TIMEOUT,
+};
+
+/** @brief Parity modes */
+enum uart_config_parity {
+	UART_CFG_PARITY_NONE,
+	UART_CFG_PARITY_ODD,
+	UART_CFG_PARITY_EVEN,
+	UART_CFG_PARITY_MARK,
+	UART_CFG_PARITY_SPACE,
+};
+
+/** @brief Number of stop bits. */
+enum uart_config_stop_bits {
+	UART_CFG_STOP_1,
+	UART_CFG_STOP_1_5,
+	UART_CFG_STOP_2,
+};
+
+/** @brief Number of data bits. */
+enum uart_config_data_bits {
+	UART_CFG_DATA_BITS_5,
+	UART_CFG_DATA_BITS_6,
+	UART_CFG_DATA_BITS_7,
+	UART_CFG_DATA_BITS_8,
+	/** 9 data bits should be used only with Mark or Space parity modes */
+	UART_CFG_DATA_BITS_9,
+};
+
+/**
+ * @brief Hardware flow control options.
+ *
+ * With flow control set to none, any operations related to flow control
+ * signals can be managed by user with uart_line_ctrl functions.
+ * In other cases, flow control is managed by hardware/driver.
+ */
+enum uart_config_flow_control {
+	UART_CFG_FLOW_CTRL_NONE,
+	UART_CFG_FLOW_CTRL_RTS_CTS,
+	UART_CFG_FLOW_CTRL_DTR_DSR,
+};
+
+/* @brief Line control signals. */
+enum uart_line_ctrl {
+	UART_LINE_CTRL_RTS,
+	UART_LINE_CTRL_DTR,
+	UART_LINE_CTRL_DCD,
+	UART_LINE_CTRL_DSR,
+};
+
+/* @brief Types of events passed to callback. */
+enum uart_event_type {
+	UART_TX_DONE,
+	UART_RX_DONE,
+	UART_ERROR,
+};
+
+/* Common communication errors for UART.*/
+enum uart_error {
+	/** @brief Overrun error */
+	UART_ERROR_OVERRUN,
+	/** @brief Parity error */
+	UART_ERROR_PARITY,
+	/** @brief Framing error */
+	UART_ERROR_FRAMING,
+	/**
+	 * @brief Break interrupt error:
+	 *
+	 * A break interrupt was received. This happens when the serial input
+	 * is held at a logic '0' state for longer than the sum of
+	 * start time + data bits + parity + stop bits.
+	 */
+	UART_ERROR_BREAK,
+};
+
+/* @brief Structure containing information about current event. */
+struct uart_event {
+	/* Type of event */
+	enum uart_event_type type;
+	/* Error type, if event is UART_ERROR, otherwise undefined. */
+	enum uart_error error;
+	/* Pointer to current transmit buffer. */
+	unsigned char *buf;
+	/* Number of bytes transmitted */
+	size_t len;
+};
+
+/**
+ * @typedef uart_callback_t
+ * @brief Define the application callback function signature for
+ * uart_set_callback() function.
```

Comment:
Did you forgot implements the **uart_set_callback** function?

Reply:
uart_set_callback is at line 510